src/vs/platform/shell/node/shellEnv.ts

```diff
@@ -72,11 +66,17 @@ export async function getResolvedShellEnv(logService: ILogService, args: NativeP
 			unixShellEnvPromise = Promises.withAsyncBody<NodeJS.ProcessEnv>(async (resolve, reject) => {
 				const cts = new CancellationTokenSource();
 
+				let timeoutValue = 10000; // default to 10 seconds
+				const configuredTimeoutValue = configurationService.getValue<unknown>('application.shellEnvironmentResolutionTimeout');
+				if (typeof configuredTimeoutValue === 'number') {
+					timeoutValue = configuredTimeoutValue * 1000 /* convert from seconds */;
```

Comment:
Maybe we `clamp(configuredTimeoutValue, 1, 120)`?

Reply:
Yeah, also updated the configuration registration with min/max.